src/components/HomePage.js

```diff
@@ -0,0 +1,64 @@
+// eslint-disable-next-line react-hooks/exhaustive-deps
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  fetchCountries,
+  fetchCountry,
+} from '../redux/countries/countriesSlice';
+import Countries from './Countries';
+import europe from '../images/europe.png';
+
+const Home = () => {
+  const [countryName, searchCountryName] = useState('');
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchCountries());
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (countryName !== '') {
+      setTimeout(() => {
+        dispatch(fetchCountry(countryName));
+      }, 1000);
+    } else {
+      dispatch(fetchCountries());
+    }
+  }, [countryName]);
```

Comment:
- [ ] You did a good job working on this milestone! 💯 However, according to the requirements:

![image](https://github.com/coker2019/metrics_app/assets/77069011/4699673b-52fa-48df-87bf-1d191a43c522)

This functionality is not working as the expected filter:

If I write "Fr" I would expect the countries to have this string in their name to show up. However, look what happens:

![image](https://github.com/coker2019/metrics_app/assets/77069011/957832e5-79a7-400b-a8bd-f47a2a2cebc3)

This is because instead of filtering the countries from the API, you are making a new request to an endpoint that sends you only one country. To fix this, I would suggest something like this:


Remove this `useEffect`:
```suggestion









```
Then, try this:

```js
// HomePage.js
const Home = () => {
  ...
  // Move the countriesArr from <Countries /> to the parent component to make it easier to filter the data:
  const countriesArr = useSelector((state) => state.countries.countriesData);
  // Filter the data based on the countryName state:
  const filteredData = countriesArr.filter((country) => country.name.toLowerCase().includes(countryName.toLowerCase()));

  return (
    <main>
      ...
      <section>
        <h3 className="country-section-title">See All Countries</h3>
        {/* Pass the filtered data as props to the component in charge of rendering the countries: */}
        <Countries countriesArr={filteredData} />
      </section>
      ...
    </main>
  )
}

```

Finally, modify your `Countries` component to receive the data as props, and that's it, now you have a stateless filter! 🤓 🔧

Reply:
well resolved! ✔️